docs/guide/vite.md

```diff
@@ -104,6 +104,209 @@ For debugging production builds, you can enable source maps to see the original
+- Use with caution as it might lead to import resolution issues if:
+  - The imported file contains `Modules/` aliased imports
+  - Deep imports within the imported file contain `Modules/` aliased imports
+- Benefits of using `@/`:
+  - More developer-friendly
+  - Enables auto-completion in supported editors
+  - More intuitive and readable paths
+- Best practices:
+  - Include mostly sources from your current build
+  - Ensure imported sources don't contain `Modules/` aliased imports when cross-importing
+
+### Editor Configuration
+
+To enable proper path resolution and autocompletion in your code editor, add a `jsconfig.json` file to your project root:
+
+```json
+{
+  "compilerOptions": {
+    "baseUrl": "./apos-build/@apostrophecms/vite/default",
+    "paths": {
+      "@/*": ["./src/*"]
+    }
+  }
+}
+```
+
+**Note:** If you've changed your project asset namespace from the default, adjust the `baseUrl` and `exclude` paths accordingly. For example, with a namespace of `my-namespace`:
+- `baseUrl` should be `./apos-build/@apostrophecms/vite/my-namespace`
+- `exclude` should include `apos-build/@apostrophecms/vite/my-namespace/dist`
+
+:::warning
+When following imports in your editor (e.g., Ctrl + Click in VSCode), you'll be taken to the `apos-build` directory rather than the original source code. This is because the `apos-build` directory contains a complete copy of your project's source code (including Admin UI) from all modules (local and npm) and serves as Vite's actual source directory for building the project.
+:::
+
+## Asset Handling
+
+The Vite integration allows direct imports of static assets like images, fonts, and other files, as well as Sass files, in both public and admin UI builds.
+
+### Importing Static Assets
+
+You can import assets directly in your JavaScript/framework code:
+
+```javascript
+// Using aliases or relative paths
+// Actual path: modules/some-module/ui/assets/logo.svg
+import logo from '@/some-module/assets/logo.svg';
+// The logo variable now contains the normalized path to the image
+```
+
+#### Framework-Specific Asset Imports
+
+**Vue**
+```vue
+<template>
+  <!-- Direct template imports -->
+  <img src="@/some-module/assets/logo.svg" alt="My logo" />
+</template>
+```
+
+**React**
+```jsx
+import logo from '@/some-module/assets/logo.svg';
+
+function MyComponent() {
+  return <img src={logo} alt="My logo" />;
+}
+```
+
+### Importing Sass
+
+You can import Sass files using standard import syntax:
+
+```scss
+/* Using aliases or relative paths */
+/* Actual path: modules/some-module/ui/scss/_styles.scss */
+@use '@/some-module/scss/styles';
+```
+
+### CSS URL Resolution
+
+There are two ways to resolve URLs in your CSS:
+
+1. **Using the Public Folder**
+```css
+/* File location: ./modules/some-module/public/font.ttf */
+@font-face {
+  font-family: MyFont;
+  src: url("/modules/some-module/font.ttf") format("truetype");
+}
+```
+
+2. **Using Source Root Path**
```

Comment:
It surprises me that this is not using `@`. I guess `src` functions like `@` for Sass for whatever reason? Can this be called out clearly if this difference is real and people need to know?

Reply:
CSS paths don't support aliases. Vite is doing all the hard job. It transforms paths that it can recognize and resolve based on a base asset path property that we provide in different environments. I don't think there is a recommended way here. You either use paths to files in your module `public/` folder (legit feature supported on both UI and back-end) or you use paths to files in your `ui/` folder (can be targeted only in the UI). 

`/src/` is the more clean way for front-end. It targets the real path inside of `apos-build` as Vite sees it. Vite then handles it properly and adds it to the manifest, which then we take and copy for asset deployment. 

`/modules/` is our "little cheat" for UI to let the assets in `module-name/public` still work as before. As a side effect we end up with 2 versions of this file, one is deployed as it is `public/.../modules/module-name/file` (because server templates might use it), and the other `public/../assets/file-123abs` (enumerated by the bundler) that is used only by the UI (be it CSS or JS). 

SASS should now understand `@/`  because it supports the Vite aliases out of the box.

Reply:
Might need help here to clarify (if needed).

Reply:
@BoDonkey TL;DR aliases are not working for CSS properties (like `url()`), absolute paths do and are normalized in production builds. Use `/modules/` if you put your stuff in `module-name/public` and when you want to access those assets from your Nunjucks templates. Use `/src/module-name/my-asset` when you want to access those files only in the front-end UI.